bot/internal/github/github.go

```diff
@@ -502,7 +502,15 @@ func (c *Client) IsOrgMember(ctx context.Context, user string, org string) (bool
 	if err != nil {
 		return false, trace.Wrap(err)
 	}
+
 	resp, err := c.client.Do(ctx, req, nil)
+	// the go-github API returns an error if the request completed
+	// succesfully but returned a non-200 response code, so we attempt
+	// to check the response before checking the error
+	if resp != nil && resp.StatusCode != 0 {
```

Comment:
This `!= 0` check looks suspicious.  Should this be `!= 200`, or is this zero condition because no response code has been specified?  If the `!= 0` is accurate a comment as to why it would be zero would be helpful.

Reply:
`StatusCode` is of type `int`, and the zero value for an `int` in Go is 0.

The if statement is attempting to make sure there is a legitimate status code to compare to, as opposed to comparing against an "unset" status code. Perhaps a little more defensive than necessary.

Reply:
I assumed it was a check for the default value, but I have never seen a library which has the response code unset on error responses.  So I just wanted to confirm this was the intention.  Thank you!

Reply:
Yeah, the behavior of this library caught me by surprise too. If there's a non 2xx response, then information is packed into the error return value instead of the response return value. 🤷‍♂️